src/main/java/com/github/sardine/impl/SardineImpl.java

```diff
@@ -1069,15 +1087,20 @@ protected HttpResponse execute(HttpRequestBase request)
 	 * @param <T> will return raw HttpResponse when responseHandler is null or value reslved using provided ResponseHandler instance
 	 * @return value resolved using response handler or raw HttpResponse when responseHandler is null
 	 */
-	protected <T> T execute(HttpClientContext context, HttpRequestBase request, ResponseHandler<T> responseHandler)
+	protected <T> T execute(HttpClientContext context, HttpUriRequestBase request, HttpClientResponseHandler<T> responseHandler)
 			throws IOException
 	{
-		HttpContext requestLocalContext = new BasicHttpContext(context);
+		Integer httpMajorVersion = (Integer) context.getAttribute(HTTP_MAJOR_VERSION);
+
+		if (httpMajorVersion != null){
+			request.setVersion(new ProtocolVersion("HTTP", httpMajorVersion, 0));
+		}
+		HttpContext requestLocalContext = new HttpClientContext(context);
```

Comment:
```suggestion
		HttpContext requestLocalContext = context;
```
removing the localContext would solve our auth problems as HttpClient5 seems to cache the previous authentication.
So we get one 401, then HttpClient selects the correct authenticator and from then, we get 2xx responses and our tests pass

Suggested change:
```java
	protected <T> T execute(HttpClientContext context, HttpUriRequestBase request, HttpClientResponseHandler<T> responseHandler)
			throws IOException
	{
		Integer httpMajorVersion = (Integer) context.getAttribute(HTTP_MAJOR_VERSION);

		if (httpMajorVersion != null){
			request.setVersion(new ProtocolVersion("HTTP", httpMajorVersion, 0));
		}
		try
		{
			if (responseHandler != null)
			{
				return this.client.execute(request, context, responseHandler);
			}
			else
			{
				return (T) this.client.execute(request, context);
			}
		}
		catch (HttpResponseException e)
		{
			// Don't abort if we get this exception, caller may want to repeat request.
			throw e;
		}
		catch (IOException e)
		{
			request.abort();
			throw e;
		}
	}
```